compiler/src/dotty/tools/dotc/interactive/Completion.scala

```diff
@@ -316,53 +323,26 @@ object Completion {
         || (mode.is(Mode.Type) && (sym.isType || sym.isStableMember))
       )
 
-    /**
-     * Find all the members of `site` that are accessible and which should be included in `info`.
-     *
-     * @param site The type to inspect.
-     * @return The members of `site` that are accessible and pass the include filter of `info`.
+    /** @param site The type to inspect.
+     *  @return The members of `site` that are accessible and pass the include filter.
      */
-    private def accessibleMembers(site: Type)(using Context): Seq[Symbol] = site match {
+    private def accessibleMembers(site: Type)(using Context): Seq[SingleDenotation] = site match {
       case site: NamedType if site.symbol.is(Package) =>
         extension (tpe: Type)
-          def accessibleSymbols = tpe.decls.toList.filter(sym => sym.isAccessibleFrom(site, superAccess = false))
+          def accessibleMembers = tpe.allMembers.toList.filter(denot => denot.symbol.isAccessibleFrom(site, superAccess = false))
```

Comment:
We can't rely on `tpe.decls` because package objects can extend traits with type parameters so, as you had pointed out before, we can't rely on symbols because then we might lose some type information. What I personally still don't understand is why for packages we should extract members of package objects in this package (`packageObjectsDecls`). There was no comment for that and all the tests still pass if I remove the special case for packages

Reply:
It looks like the old implementation was not only unnecessarily complicated but also buggy. I simplified it so that `allMembers` is not necessary